### ISSUE TITLE
Warning: componentWillMount has been renamed, and is not recommended for use.

### ISSUE THREAD

**Issue by greenpau** [wictorwilen/msteams-react-base-component#2]
The impacted code is: `./node_modules/msteams-react-base-component/lib/index.js.TeamsBaseComponent.render`

When rendering the page, I see the following warning:

```
react-dom.development.js:88 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components:

...


./node_modules/msteams-react-base-component/lib/index.js.TeamsBaseComponent.render | @ | index.js:98
```

React does not like the following:

```js
    /**
     * Static method to render the component
     * @param element DOM element to render the control in
     * @param props Properties
     */
    TeamsBaseComponent.render = function (element, props) {
        react_dom_1.render(React.createElement(this, props), element);
    };
```

**Comment by wictorwilen** [comment]
This is a known issue and will be fixed in future updates when moving to React Hooks.  For now you can just ignore it as it's only a warning.

**Comment by wictorwilen** [comment]
The new `3.0.0-preview` version is now using React hooks which removes this error. Feel free to try that one out.

**Comment by wictorwilen** [comment]
This is now included in the published `3.0.0`
